FirebaseRemoteConfig/Sources/RCNPersonalization.m

```diff
@@ -37,16 +38,35 @@ - (void)logArmActive:(NSString *)key config:(NSDictionary *)config {
   }
 
   NSDictionary *metadata = ids[key];
-  if (!metadata || metadata[kPersonalizationId] == nil) {
+  if (!metadata) {
     return;
   }
 
+  NSString *choiceId = metadata[kChoiceId];
+  if (choiceId == nil) {
+    return;
+  }
+
+  // This gets dispatched to a serial queue, so this is OK. But even if not, it'll just possibly
+  // log more.
+  if (self->_armsCache[key] == choiceId) {
+    return;
+  }
+  self->_armsCache[key] = choiceId;
+
   [self->_analytics logEventWithOrigin:kAnalyticsOriginPersonalization
                                   name:kAnalyticsPullEvent
                             parameters:@{
-                              kArmKey : metadata[kPersonalizationId],
-                              kArmValue : values[key].stringValue
+                              kArmKey : key,
+                              kArmValue : values[key].stringValue,
+                              kPersonalizationIdLogKey : metadata[kPersonalizationId],
+                              kArmIndexLogKey : metadata[kArmIndex],
+                              kGroup : metadata[kGroup]
```

Comment:
Thinking: the [design](http://go/sage-p13n-bq) lists five things we want to log:
1. Developer friendly identifier of the personalization (either the rc param, or an id customers can map back to rc param), satisfied by the `arm_key` field
2. The choice that was made (a string value), satisfied by the `arm_value` field
3. The group the user was in (“p13n” or “baseline”), satisfied by the `group` field
4. Personalization ID, satisfied by the `personalization_id` field
5. Arm index, satisfied by the `arm_index` field

Thinking: the external event name is arbitrary, we just need to document what it is. `personalization_assignment` seems self-descriptive to me 👍

Minor: renaming `arm_key` to something like `rc_parameter` would be more self-descriptive given it communicates an RC parameter. Likewise, renaming the `key` arg passed to `logArmActive` to `rcParameter` would clarify its purpose.